### PR TITLE
Trac role to GitHub

### PR DESCRIPTION
**Pull request by roed314**
### 📚 Description

Changes the `:trac:` role to point to github instead, and changes deprecation and stopgap warnings.  Changes all the resulting urls that are printed in doctests, as well as various comments in code.

Does not update the developer documentation to explain how to use github.

### CODE REVIEW

File: build/pkgs/ecm/spkg-install.in [hunk 1]
```diff
@@ -92,10 +92,10 @@ else
             # Don't add debug symbols because configure otherwise
             # fails due to a bus error in Apple's 'ld' when trying
             # to determine if global symbols are prefixed with an
-            # underscore (cf. http://trac.sagemath.org/sage_trac/ticket/5847#comment:35 ff.):
+            # underscore (cf. https://github.com/sagemath/sage/issues/5847 ff.):
```
Comment (mkoeppe) [inline build/pkgs/ecm/spkg-install.in]: You can keep the fragments in the URL. We made sure that they also work.

File: build/pkgs/sagelib/spkg-install [hunk 2]
```diff
@@ -29,7 +29,7 @@ fi
 ## Building takes places in the build/ subdirectory.
 ##
 ## As a special exception, we feed SAGE_PKGS.
-## They are needed by src/sage/misc/package.py.  See meta-ticket #28815 for planned changes to this.
+## They are needed by src/sage/misc/package.py.  See github issue #28815 for planned changes to this.
```
Comment (mkoeppe) [inline build/pkgs/sagelib/spkg-install:32]: better as URL?

File: src/doc/en/developer/git_trac.rst [hunk 3]
```diff
@@ -76,8 +76,8 @@ to the Sage directory and tell ``git trac`` about your trac account::
 
     [user@localhost sage]$ git trac config --user USERNAME --pass 'PASSWORD'
     Trac xmlrpc URL:
-        https://trac.sagemath.org/xmlrpc (anonymous)
```
Comment (mkoeppe) [inline src/doc/en/developer/git_trac.rst]: The changes in this file are best reverted.

File: src/doc/en/developer/trac.rst [hunk 4]
```diff
@@ -280,7 +280,7 @@ To produce the warning message, use code like the following:
         TICKET_NUM)
```
Comment (mkoeppe) [inline src/doc/en/developer/trac.rst]: Best to revert the changes to this file?

File: src/sage/misc/superseded.py [hunk 5]
```diff
@@ -5,16 +5,15 @@
 add a deprecation warning. This will be shown once, the first time
 that the deprecated function is called.
 
-Note that all doctests in the following use the trac ticket number
-:trac:`13109`, which is where this mandatory argument to
+Note that all doctests in the following use the github issue number
+:github:`13109`, which is where this mandatory argument to
```
Comment (mkoeppe) [inline src/sage/misc/superseded.py]: Does the `:github:` role already exist?

File: src/sage/rings/function_field/order.py [hunk 6]
```diff
@@ -1773,7 +1773,7 @@ def decomposition(self, ideal):
         finite-dimensional algebra, a construct for which we do
         support primary decomposition.
 
-        See :trac:`attachment/ticket/28094/decomposition.pdf`
+        See :trac:`28094` and https://github.com/sagemath/sage-prod/files/10659303/decomposition.pdf.gz
```
Comment (mkoeppe) [inline src/sage/rings/function_field/order.py]: You can drop the `-prod` in the URL.

File: src/sage/rings/qqbar.py [hunk 7]
```diff
@@ -730,7 +730,7 @@ def _factor_multivariate_polynomial(self, f, proof=True):
         REFERENCES:
 
         - [GCL1992]_ Section 8.8
-        - https://trac.sagemath.org/attachment/ticket/25390/qqbar.pdf
+        - :trac:`25390` and https://github.com/sagemath/sage-prod/files/10659152/qqbar.pdf.gz
```
Comment (mkoeppe) [inline src/sage/rings/qqbar.py:734]: Drop `-prod`

File: src/sage/schemes/elliptic_curves/ell_field.py [hunk 8]
```diff
@@ -988,7 +988,7 @@ def division_field(self, l, names='t', map=False, **kwds):
 
         verbose("Adjoining Y-coordinates of %s-torsion points" % l)
 
-        # THEOREM (Cremona, http://trac.sagemath.org/ticket/11905#comment:21).
+        # THEOREM (Cremona, https://github.com/sagemath/sage/issues/11905).
         # Let K be a field, E an elliptic curve over K and p an odd
```
Comment (mkoeppe) [inline src/sage/schemes/elliptic_curves/ell_field.py:992]: keep the fragment

**Comment by kwankyu** [comment]
How about introducing a new role `:gh:` to replace the role of `:trac:` at this time? Or later in a new PR?

We may keep both for some time.

**Comment by tobiasdiez** [comment]
Maybe a general purpose `issue` role, so that we don't have to change everything in a few years when we migrate to the next hip and happening platform?

**Comment by kwankyu** [comment]
Never in a few years :-) 

Like `:trac:`, I think that specificity of `:gh:` would explain the role more than a general purpose name. And it's short!

File: build/pkgs/ecm/spkg-install.in (hunk 1 above)
Comment (roed314) [reply build/pkgs/ecm/spkg-install.in]: Oh great!  I'll go back and undo those changes.

File: build/pkgs/sagelib/spkg-install (hunk 2 above)
Comment (roed314) [reply build/pkgs/sagelib/spkg-install:32]: I agree that it would probably be better as a URL, but there were enough of these changes that I didn't want to include that as part of this PR.

File: src/doc/en/developer/trac.rst (hunk 4 above)
Comment (roed314) [reply src/doc/en/developer/trac.rst]: Agreed.

File: src/sage/misc/superseded.py (hunk 5 above)
Comment (roed314) [reply src/sage/misc/superseded.py]: I added it as part of this PR (by changing `src/tox.ini` and `src/sage/misc/sagedoc.py`), but based on the discussion below and consulting with some people at SD117, I'm changing it to `:issue:`.

File: src/sage/rings/function_field/order.py (hunk 6 above)
Comment (roed314) [reply src/sage/rings/function_field/order.py]: Will do.  I'll note that the attachment links right now on issues like #28094 point to the `sage-prod` version of the url.

File: src/sage/schemes/elliptic_curves/ell_field.py (hunk 8 above)
Comment (roed314) [reply src/sage/schemes/elliptic_curves/ell_field.py:992]: I've added all the comments back in.

**Comment by roed314** [comment]
I like @tobiasdiez suggestion of `:issue:`, since I think it explains the role better than either `:github:` or `:gh:`.  I also find `:gh:` a bit opaque.  Of course, others are welcome to offer their opinion.

I think we should the `:trac:` role for quite a while, both to ease the transition of developers and since changing it would require more than 4000 changes to the sage library.

File: src/sage/rings/qqbar.py (hunk 7 above)
Comment (kwankyu) [reply src/sage/rings/qqbar.py:734]: @mkoeppe On the other side, could we remove `-prod` in the attachment links in the converted issues?

**Comment by kwankyu** [comment]
`:issue:` may be a bit misleading when the target is a PR.

**Comment by roed314** [comment]
A bit, but the link will still work.  I also think it will help encourage people to link to the issue where discussion of the problems occurs (which I think is usually preferable when both issue and PR exist).

**Comment by kwankyu** [comment]
I am okay with ":issue:". Let's have the sphinx role in quickly.

File: src/doc/en/developer/git_trac.rst (hunk 3 above)
Comment (roed314) [reply src/doc/en/developer/git_trac.rst]: Done.

File: src/sage/rings/qqbar.py (hunk 7 above)
Comment (mkoeppe) [reply src/sage/rings/qqbar.py:734]: That's now https://github.com/sagemath/trac-to-github/issues/188

File: build/pkgs/glpk/spkg-install.in [hunk 9]
```diff
@@ -1,6 +1,6 @@
 cd src/
 
-# Use newer version of config.guess and config.sub (see Github issue #19713)
+# Use newer version of config.guess and config.sub (see Trac #19713)
 cp "$SAGE_ROOT"/config/config.* .
```
Comment (mkoeppe) [inline build/pkgs/glpk/spkg-install.in:4]: Trac snuck back in here

**Comment by mkoeppe** [comment]
One test failure:
```
sage -t --random-seed=214214552109874894428970580105012371878 sage/misc/superseded.py
**********************************************************************
File "sage/misc/superseded.py", line 52, in sage.misc.superseded._check_issue_number
Failed example:
    _vcheck_issue_number(int(10))
Exception raised:
    Traceback (most recent call last):
      File "/__w/sage/sage/src/sage/doctest/forker.py", line 695, in _run
        self.compile_and_execute(example, compiler, test.globs)
      File "/__w/sage/sage/src/sage/doctest/forker.py", line 1093, in compile_and_execute
        exec(compiled, globs)
      File "<doctest sage.misc.superseded._check_issue_number[3]>", line 1, in <module>
        _vcheck_issue_number(int(Integer(10)))
    NameError: name '_vcheck_issue_number' is not defined
```

File: build/pkgs/glpk/spkg-install.in (hunk 9 above)
Comment (roed314) [reply build/pkgs/glpk/spkg-install.in:4]: Fixed, thanks.

File: src/doc/en/developer/advanced_git.rst [hunk 10]
```diff
@@ -11,8 +11,8 @@ Advanced Git
     **Sage development is scheduled to move to GitHub in February 2023.** The exact
     date will be announced in `<https://groups.google.com/g/sage-devel>`_. After
     the transition, some parts of this guide (especially those related with `the
-    Sage Trac server <https://trac.sagemath.org>`_) will become obsolete and be
-    updated according to the new workflow on GitHub. See our `transition guide from Trac to
+    Sage Issue server <https://trac.sagemath.org>`_) will become obsolete and be
+    updated according to the new workflow on GitHub. See our `transition guide from Issue to
```
Comment (mkoeppe) [inline src/doc/en/developer/advanced_git.rst]: This one should be reverted

File: src/doc/en/developer/git_trac.rst [hunk 11]
```diff
@@ -3,16 +3,16 @@
 .. _chapter-git_trac:
 
 ====================================
-Optional: Using the Git-Trac Command
+Optional: Using the Git-Issue Command
```
Comment (mkoeppe) [inline src/doc/en/developer/git_trac.rst]: Changes in this file should be reverted

File: src/doc/en/developer/manual_git.rst [hunk 12]
```diff
@@ -3,16 +3,16 @@
 .. _chapter-manual-git:
 
 ===================================
-Using Git with the Sage Trac Server
```
Comment (mkoeppe) [inline src/doc/en/developer/manual_git.rst]: Changes in this file should be reverted

File: src/doc/en/developer/packaging.rst [hunk 13]
```diff
@@ -1079,7 +1079,7 @@ For Python packages available from PyPI, there is another shortcut::
     [...............................................................]
 
 The ``upstream_url`` information serves yet another purpose.
-Developers who wish to test a package update from a Trac branch before
+Developers who wish to test a package update from a Issue branch before
 the archive is available on a Sage mirror can do so by configuring
```
Comment (mkoeppe) [inline src/doc/en/developer/packaging.rst]: Change "Issue branch" to "PR"

File: src/doc/en/developer/reviewer_checklist.rst [hunk 14]
```diff
@@ -11,8 +11,8 @@ The reviewer's check list
     **Sage development is scheduled to move to GitHub in February 2023.** The exact
     date will be announced in `<https://groups.google.com/g/sage-devel>`_. After
     the transition, some parts of this guide (especially those related with `the
-    Sage Trac server <https://trac.sagemath.org>`_) will become obsolete and be
```
Comment (mkoeppe) [inline src/doc/en/developer/reviewer_checklist.rst]: Revert the changes in this file

File: src/doc/en/developer/sage_manuals.rst [hunk 15]
```diff
@@ -107,7 +107,7 @@ by Sage, you can link toward it without specifying its full path:
    :widths: 4 4 4
    :header-rows: 0
 
-   * - Trac server
+   * - Issue server
```
Comment (mkoeppe) [inline src/doc/en/developer/sage_manuals.rst]: Revert

File: src/doc/en/developer/trac.rst [hunk 16]
```diff
@@ -3,16 +3,16 @@
 .. _chapter-sage-trac:
 
 ====================
-The Sage Trac Server
+The Sage Issue Server
 ====================
```
Comment (mkoeppe) [inline src/doc/en/developer/trac.rst]: Revert changes in this file

File: src/doc/en/developer/walk_through.rst [hunk 17]
```diff
@@ -11,8 +11,8 @@ Sage Development Process
     **Sage development is scheduled to move to GitHub in February 2023.** The exact
     date will be announced in `<https://groups.google.com/g/sage-devel>`_. After
     the transition, some parts of this guide (especially those related with `the
-    Sage Trac server <https://trac.sagemath.org>`_) will become obsolete and be
```
Comment (mkoeppe) [inline src/doc/en/developer/walk_through.rst]: Revert changes here

File: src/doc/en/developer/workflows.rst [hunk 18]
```diff
@@ -11,8 +11,8 @@ Distributed Development
     **Sage development is scheduled to move to GitHub in February 2023.** The exact
     date will be announced in `<https://groups.google.com/g/sage-devel>`_. After
     the transition, some parts of this guide (especially those related with `the
-    Sage Trac server <https://trac.sagemath.org>`_) will become obsolete and be
-    updated according to the new workflow on GitHub. See our `transition guide from Trac to
+    Sage Issue server <https://trac.sagemath.org>`_) will become obsolete and be
```
Comment (mkoeppe) [inline src/doc/en/developer/workflows.rst]: Revert

File: src/doc/en/developer/workflows.rst [hunk 19]
```diff
@@ -11,8 +11,8 @@ Distributed Development
...
+    updated according to the new workflow on GitHub. See our `transition guide from Issue to
     GitHub
```
Comment (mkoeppe) [inline src/doc/en/developer/workflows.rst]: Revert

File: src/doc/en/developer/workspace.rst [hunk 20]
```diff
@@ -38,7 +38,7 @@ For this to work, you first have to authorize Gitpod with Trac:
  2. Save the private key as a secure environment variable in Gitpod using
     ``gp env PRIVATE_SSH_KEY="$(<tempkey)"``,
     or by using the `Gitpod UI <https://www.gitpod.io/docs/environment-variables#using-the-account-settings>`_.
- 3. Register the public key with Trac following the instructions in :ref:`section-trac-ssh-key`.
+ 3. Register the public key with Issue following the instructions in :ref:`section-trac-ssh-key`.
  4. Close this Gitpod workspace.
```
Comment (mkoeppe) [inline src/doc/en/developer/workspace.rst]: Revert

File: src/doc/en/installation/linux.rst [hunk 21]
```diff
@@ -21,5 +21,5 @@ distribution, consider upgrading your distribution.
 Gentoo users might want to give a try to
 `sage-on-gentoo <https://github.com/cschwan/sage-on-gentoo>`_.
 
-The  :trac:`Trac wiki page Distribution <wiki/Distribution>` collects information
+The  :trac:`Issue wiki page Distribution <wiki/Distribution>` collects information
```
Comment (mkoeppe) [inline src/doc/en/installation/linux.rst]: Need a separate role for links to the new GH wiki

**Comment by roed314** [comment]
Sorry about that; I had meant to just avoid the `doc` directory in this PR, but accidentally ran `sed` one level too high in the source tree.  I think the relevant files should be reverted now.

File: src/doc/en/installation/linux.rst (hunk 21 above)
Comment (roed314) [reply src/doc/en/installation/linux.rst]: True, though I'm just going to leave the doc folder alone for now.

File: src/doc/en/developer/packaging.rst (hunk 13 above)
Comment (roed314) [reply src/doc/en/developer/packaging.rst]: I'm leaving this untouched for now (we need to update documentation in a separate PR).

File: build/pkgs/ecl/SPKG.rst [hunk 22]
```diff
@@ -43,7 +43,7 @@ Special Update/Build Instructions
 
 -  Do NOT quote SAGE_LOCAL when setting CPPFLAGS and/or LDFLAGS,
    in spkg-install as this caused the build to break. See
-   http://trac.sagemath.org/sage_trac/ticket/10187#comment:117
+   https://github.com/sagemath/sage/issues/10187#comment:117
```
Comment (mkoeppe) [inline build/pkgs/ecl/SPKG.rst:47]: I think this needs to be changed to an `:issue:`, or Sphinx will complain.

Comment (mkoeppe) [reply build/pkgs/ecl/SPKG.rst:47]: see https://github.com/sagemath/sage/actions/runs/4137167886/jobs/7151994051#step:6:21601

File: build/pkgs/ecl/SPKG.rst [hunk 23]
```diff
@@ -43,7 +43,7 @@ Special Update/Build Instructions
 
 -  Do NOT quote SAGE_LOCAL when setting CPPFLAGS and/or LDFLAGS,
    in spkg-install as this caused the build to break. See
-   http://trac.sagemath.org/sage_trac/ticket/10187#comment:117
+   :issue:`10187`#comment:117
```
Comment (mkoeppe) [inline build/pkgs/ecl/SPKG.rst]: Pretty sure the # has to be inside the backticks

Comment (roed314) [reply build/pkgs/ecl/SPKG.rst]: Thanks

File: build/pkgs/ecl/SPKG.rst (hunk 22 above)
Comment (roed314) [reply build/pkgs/ecl/SPKG.rst:47]: Waiting to see the results of the docbuild....

**Review by mkoeppe — APPROVED** [review]
Looking good.

**Comment by dimpase** [comment]
there is still a number of explicit (rewritten to URLs) trac.sagemath.org/ticket/<...> links in docstrings which should become :trac:`<...>`.
(in another PR)

**Comment by dimpase** [comment]
@vbraun - can you merge this soon? It's touching a lo of files for a good reason, and having it linger on
is entering a rebase hell...

**Comment by mkoeppe** [comment]
I think the best way to request this is to mark with a high-priority label
